pkg/api/api.go

```diff
@@ -453,6 +453,7 @@ func (hs *HTTPServer) registerRoutes() {
 
 		// Dashboard
 		apiRoute.Group("/dashboards", func(dashboardRoute routing.RouteRegister) {
+			dashboardRoute.Get("/", authorize(ac.EvalPermission(dashboards.ActionDashboardsRead)), routing.Wrap(hs.GetDashboards))
```

Comment:
any reason to add this vs just keep using the one that is already exposed in search?

If we were not actively looking at how to replace/migrate these to k8s flavor, adding new endpoints would make sense.

Reply:
the search query first fetches the list of the dashboard the user has access to:
https://github.com/grafana/grafana/blob/f3d742caeb32d2a7e60df64744ece3063b8e2fac/pkg/services/sqlstore/permissions/dashboard.go#L178-L203
and then filters out by the folder under the dashboards/folders live.

with nested folders enabled, where subfolders inherit permissions from other folders finding all the resources the user has permission can be expensive.

this endpoint behaves differently:
- for listing the dashboards under a folder: it checks whether the user has access to the folder (and hence to the resources under the folder) and then fetches the dashboards under the folder
- for listing root folders: it fetches the dashboard under the root and filters out those that the user does not have access to

can you share a bit more about the implications that would introduce for the k8s?

Reply:
do we need both approaches available at the same time?  could it be a query param or feature flag on the existing search endpoint?

> can you share a bit more about the implications that would introduce for the k8s?

Mostly a concern on having multiple ways to get to the same thing, and introducing more things that will immediately need a migration path.